addons/assignGear/functions/fnc_getLoadoutFromConfig.sqf

```diff
@@ -0,0 +1,230 @@
+/*
+ * Author: PabstMirror
+ * Gets a loadout array from a loadout config path
+ *
+ * Arguments:
+ * 0: Config Path <CONFIG>
+ *
+ * Return Value:
+ * <ARRAY>
+ *
+ * Example:
+ * [missionConfigFile >> "CfgLoadouts" >> (faction player) >> (typeOf player)] call potato_assignGear_fnc_getLoadoutFromConfig
+ */
+#include "script_component.hpp"
+
+params ["_path"];
+TRACE_1("params",_path);
+
+private _configUniform = getArray (_path >> "uniform");
+private _configVest = getArray (_path >> "vest");
+private _configHeadgear = getArray (_path >> "headgear");
+private _configBackpack = getArray (_path >> "backpack");
+private _configBackpackItems = getArray (_path >> "backpackItems");
+private _configWeapons = getArray (_path >> "weapons");
+private _configLaunchers = getArray (_path >> "launchers");
+private _configHandguns = getArray (_path >> "handguns");
+private _configMagazines = getArray (_path >> "magazines");
+private _configItems = getArray (_path >> "items");
+private _configLinkedItems = getArray (_path >> "linkedItems");
+private _configAttachments = getArray (_path >> "attachments");
+private _configSecondaryAttachments = getArray (_path >> "secondaryAttachments");
+
+private _uniformArray = [];
+private _vestArray = [];
+private _backpackArray = [];
+private _containersArray = [[0],[0],[0]];
+
+if (!(_configUniform isEqualTo [])) then {
+    private _uniform = selectRandom _configUniform;
+    private _container = getText (configFile >> "CfgWeapons" >> _uniform >> "ItemInfo" >> "containerClass");
+    private _uniformSpace = getNumber (configFile >> "CfgVehicles" >> _container >> "maximumLoad");
+    TRACE_3("",_uniform,_container,_uniformSpace);
+    _uniformArray = [_uniform, []];
+    _containersArray set [0, [_uniformSpace, _uniformArray]];
+};
+if (!(_configVest isEqualTo [])) then {
+    private _vest = selectRandom _configVest;
+    private _container = getText (configFile >> "CfgWeapons" >> _vest >> "ItemInfo" >> "containerClass");
+    private _vestSpace = getNumber (configFile >> "CfgVehicles" >> _container >> "maximumLoad");
+    TRACE_3("",_vest,_container,_vestSpace);
+    _vestArray = [_vest, []];
+    _containersArray set [1, [_vestSpace, _vestArray]];
+};
+if (!(_configBackpack isEqualTo [])) then {
+    private _backpack = selectRandom _configBackpack;
+    private _backpackSpace = getNumber (configFile >> "CfgVehicles" >> _backpack >> "maximumLoad");
+    TRACE_2("",_backpack,_backpackSpace);
+    _backpackArray = [_backpack, []];
+    _containersArray set [2, [_backpackSpace, _backpackArray]];
+};
+
+TRACE_1("containers",_containersArray);
+
+private _headgear = if (_configHeadgear isEqualTo []) then {""} else {selectRandom _configHeadgear};
+private _binocularArray = [];
+private _assignedItems = ["", "", "", "", "", ""];
+
+private _fnc_addItemsToContainer = {
```

Comment:
Should these private funcs be externalized? This function will be called a lot, and we're wasting some IO re-deffing them each call